src/Router/Zf2.php

```diff
@@ -19,17 +19,14 @@
  *
  * This router implementation consumes zend-mvc's TreeRouteStack, (the default
  * router implementation in a ZF2 application). The addRoute() method injects
- * segment routes into the TreeRouteStack, and manages an internal route stack
- * in order to do HTTP method negotiation after a successful match (as the ZF2
- * "Method" router implementation will return a result indistinguishable from a
- * 404 otherwise).
+ * segment routes into the TreeRouteStack, and create a fail route for HTTP
+ * method negotiation (as the ZF2 "Method" router implementation will return
+ * a result indistinguishable from a 404 otherwise).
  */
 class Zf2 implements RouterInterface
 {
-    /**
-     * @var Route[] Registered routes
-     */
-    private $routes = [];
+    // Name of the route fail
+    const ROUTE_FAIL = 'fail';
```

Comment:
Let's use the name and value "HTTP_METHOD_NOT_ALLOWED" here, to make it explicit what it's for.